cmake/SetupJemalloc.cmake

```diff
@@ -29,33 +29,20 @@ get_filename_component(
   ${JEMALLOC_LIBRARIES}
   NAME
   )
-get_filename_component(
-  JEMALLOC_LIB_TYPE
-  ${JEMALLOC_LIB_NAME}
-  LAST_EXT
-  )
-while(NOT "${JEMALLOC_LIB_TYPE}" STREQUAL ".so"
-    AND NOT "${JEMALLOC_LIB_TYPE}" STREQUAL ".a"
-    AND NOT "${JEMALLOC_LIB_TYPE}" STREQUAL ".dylib")
-  get_filename_component(
-    JEMALLOC_LIB_NAME
-    ${JEMALLOC_LIB_NAME}
-    NAME_WLE
-    )
-  get_filename_component(
-    JEMALLOC_LIB_TYPE
-    ${JEMALLOC_LIB_NAME}
-    LAST_EXT
-    )
-endwhile()
-
-if("${JEMALLOC_LIB_TYPE}" STREQUAL ".a")
+if(APPLE)
+  string(FIND "${JEMALLOC_LIB_NAME}" ".dylib" FOUND_SHARED)
+else()
+  string(FIND "${JEMALLOC_LIB_NAME}" ".so" FOUND_SHARED)
```

Comment:
Can you use `string(REGEX MATCH ... ".so$" ...)` instead to make sure this is the last extension? If this is more complicated for some reason, please add a code comment.